block_server.py

```diff
@@ -483,36 +363,33 @@ def _start_config_iocs(self):
         # Start the IOCs, if they are available and if they are flagged for autostart
         # Note: autostart means the IOC is started when the config is loaded,
         # restart means the IOC should automatically restart if it stops for some reason (e.g. it crashes)
-        for n, ioc in self._active_configserver.get_all_ioc_details().iteritems():
+        for name, ioc in self._active_configserver.get_all_ioc_details().iteritems():
             try:
                 # IOCs are restarted if and only if auto start is True. Note that auto restart instructs proc serv to
                 # restart an IOC if it terminates unexpectedly and does not apply here.
                 if ioc.autostart:
                     # Throws if IOC does not exist
-                    running = self._ioc_control.get_ioc_status(n)
-                    if running == "RUNNING":
-                        # Restart it
-                        self._ioc_control.restart_ioc(n)
-                    else:
-                        # Start it
-                        self._ioc_control.start_ioc(n)
+                    self._ioc_control.restart_ioc(name) \
+                        if self._ioc_control.get_ioc_status(name) == "RUNNING" \
+                        else self._ioc_control.start_ioc(name)
             except Exception as err:
-                print_and_log("Could not (re)start IOC %s: %s" % (n, str(err)), "MAJOR")
+                print_and_log("Could not (re)start IOC {}: {}".format(name, err), "MAJOR")
 
         # Give it time to start as IOC has to be running to be able to set restart property
         sleep(2)
-        for n, ioc in self._active_configserver.get_all_ioc_details().iteritems():
+        for name, ioc in self._active_configserver.get_all_ioc_details().iteritems():
             if ioc.autostart:
                 # Set the restart property
-                print_and_log("Setting IOC %s's auto-restart to %s" % (n, ioc.restart))
-                self._ioc_control.set_autorestart(n, ioc.restart)
+                print_and_log("Setting IOC %s's auto-restart to %s" % (name, ioc.restart))
+                self._ioc_control.set_autorestart(name, ioc.restart)
 
-    def _get_iocs(self, include_running=False):
+    def _get_iocs(self):
```

Comment:
Instead of deleting db client why don't we move this whole function into it and out of the block server main code.